### ISSUE TITLE
Unpacking equations term by term

### ISSUE THREAD

**Issue by glugeorge** [glugeorge/coupled_ice_hydrology#2]
@jkingslake
Plotting equations term by term yields these initial plots:
Initial steady state condition:
<img width="1067" alt="image" src="https://user-images.githubusercontent.com/25989736/200054836-5491a87d-27e3-43b6-869c-9a71f96be5e0.png">

Final steady state after perturbation to A leading to retreat of grounding line:
<img width="1080" alt="image" src="https://user-images.githubusercontent.com/25989736/200055021-e8b5b6b7-fa72-4167-baf2-48271e5696bc.png">

Plotting initial and final against each other (legends are a bit messy, need to think of how to better display):
<img width="1135" alt="image" src="https://user-images.githubusercontent.com/25989736/200055379-5d1e03a4-2002-493e-9f66-3a0cafc23ab1.png">

This plot below shows how the advection term is largely ineffective until near the terminus. 
<img width="379" alt="image" src="https://user-images.githubusercontent.com/25989736/200055747-d3d6e24d-2813-4b6f-9436-6db06b0e0555.png">

**Comment by glugeorge** [comment]
I think the most interesting plot here is how the dN/dx term and the effective pressure interact to balance to define the f(Q,S) term

**Comment by jkingslake** [comment]
OK, nice. Are all these dimensional variables?

I see a bit of a blip at the place the ice-flow grid changes. Any ideas what is causing that? 

Agreed the bottom-left plots are really interesting.

**Comment by glugeorge** [comment]
Yeah these variables are dimensional. I'm trying to think of how to best label the y axis - should I just use the units for the equations? 

The blip I believe is due to errors coming from interpolating grids and also taking the gradient of the grids. Here is grad(x):
<img width="521" alt="image" src="https://user-images.githubusercontent.com/25989736/200063576-c505b480-aa35-440f-8e00-40c5dd12bf9f.png">

Here is grad(hu):
<img width="491" alt="image" src="https://user-images.githubusercontent.com/25989736/200063803-1185aa87-8a81-4482-a407-82e55f8de0ed.png">

**Comment by glugeorge** [comment]
<img width="1187" alt="image" src="https://user-images.githubusercontent.com/25989736/201227756-da8c26b1-cacc-451c-81e8-003b7a4bbe0c.png">
Using 10% of the grid being at a finer resolution (with 300 data points in the fine domain, 100 data points in coarse domain). We can see that the overall shapes of the curves remain the same whilst the interpolation errors move further up the glacier. There is some weird oscillations in the bottom left plot - I think these are some artifacts coming out of the solver, which solved "with potential inaccuracies". Will continue trying additional simulations at finer resolutions

**Comment by jkingslake** [comment]
How long did the simulation take?

**Comment by glugeorge** [comment]
<img width="1140" alt="Screen Shot 2022-11-12 at 16 30 05" src="https://user-images.githubusercontent.com/25989736/201496033-766f26fb-ab6d-4ff1-a097-65b8c1eec75a.png">
This one with 900 grid points (800 on fine resolution for 25% of the grid) took 77 minutes. Overall it has the same observed behavior as the other resolution simulations, except the advection term effect becomes even smaller. I'd trust this one more than my initial run of 100 on fine resolution for 3% (even though the resolution becomes a bit worse in fine domain for this new one) because in that first run, the discontinuity happens right in the same area where the advection term becomes important

**Comment by jkingslake** [comment]
Looks good. I think the final versions of these runs may need to use a uniform resolution. This blip at the junction between the two gross is not very satisfactory.  But continue using a lower resolution for now and then increase later for the final versions. Also it will be important to do that convergence analysis we have talked about, where you record some metric of the model (like GL position) for many simulations using increasingly fine resolution. 

Do you have the accompanying plot of the variables for this simulation?

**Comment by jkingslake** [comment]
I think it could be useful for the discussion to describe three or four different regions, based on what the dominant terms are.

**Comment by glugeorge** [comment]
<img width="539" alt="image" src="https://user-images.githubusercontent.com/25989736/201500157-65d44d67-a73d-4c08-a65f-ec37605a8d2d.png">
Here are variables for this simulation. And sure, I can do this convergence analysis with the steady state solutions. Yeah, the change between the two grids is pretty annoying, and I think it's mostly in instances where we have grad(variable)/grad(grid) for the term. I'll try playing around with ways of getting rid of this blip

I'll also try to describe these regions
